infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerContainers.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.workspace.infrastructure.docker;
+
+import com.google.common.base.Strings;
+
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InternalInfrastructureException;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.ContainerInfo;
+import org.eclipse.che.plugin.docker.client.json.ContainerListEntry;
+import org.eclipse.che.plugin.docker.client.params.ListContainersParams;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Helps to work with docker containers in infrastructure domain context. */
+@Singleton
+public class DockerContainers {
+
+    private final DockerConnector docker;
+
+    @Inject
+    public DockerContainers(DockerConnector docker) {
+        this.docker = docker;
+    }
+
+    /**
+     * Lookups all containers owned by runtime with given identity.
+     *
+     * <p>Note that the operation makes 1 + N calls using {@link DockerConnector},
+     * first it searches for all the running containers and then inspects each of them directly,
+     * apparently response time depends on {@link DockerConnector} capabilities.
+     *
+     * @param id
+     *         identity of runtime which owns containers
+     * @return list of running containers owned by runtime with given id
+     * @throws InternalInfrastructureException
+     *         when any error occurs during lookup
+     */
+    public List<ContainerInfo> find(RuntimeIdentity id) throws InternalInfrastructureException {
+        try {
+            List<ContainerInfo> result = new ArrayList<>();
+            for (ContainerListEntry runningEntry : docker.listContainers(ListContainersParams.create().withAll(false))) {
```

Comment:
No, i have not, but obviously it should improve the the time of selection. Thank you!